### PR TITLE
Wip/alarmfixes

### PR DESCRIPTION
**Pull request by sulkaharo**
Apologies for large amount of unrelated changes in one PR, the huge amount of recent changes made the rebasing a bit of a challenge. ;)

Includes first pass of time zone support for profiles - now requires "timezone" variable defined in the profile object.

Show at least a part of the notification message in client upon alarm (not ideal solution, but already better than not seeing any message). 

Fixes a double conversion of SGV in ar2.js.

Optionally warn user upon cannula aging, configurable using CAGE_WARN and CAGE_URGENT.

@jasoncalabrese especially the time zone stuff needs testing. Works for us, but...

### CODE REVIEW

File: lib/profilefunctions.js [hunk 1]
```diff
@@ -36,30 +37,41 @@ function init(profileData) {
 	
 	profile.getValueByTime = function getValueByTime (time, valueContainer) {
       if (!time) { time = new Date(); }
+      var tz = 'UTC';
+      if (profile.getTimezone()) tz = profile.getTimezone();
```
Comment (jasoncalabrese) [inline lib/profilefunctions.js]: Codacy Issue found: [Expected '{' and instead saw 'tz'.](https://www.codacy.com/app/Nightscout/cgm-remote-monitor/file/768506899/issues/source?bid=2452379&fileBranchId=2457543#l41)

File: lib/plugins/cannulaage.js [hunk 2]
```diff
@@ -12,40 +12,83 @@ function init() {
...
-              message = treatment.notes;
+              returnValue.message = treatment.notes;
             }
           }
         }
       }
     });
+    
+    return returnValue;
+  }
```
Comment (jasoncalabrese) [inline lib/plugins/cannulaage.js]: Codacy Issue found: [Missing semicolon.](https://www.codacy.com/app/Nightscout/cgm-remote-monitor/file/768506884/issues/source?bid=2452379&fileBranchId=2457543#l75)

File: static/js/client.js [hunk 3]
```diff
@@ -103,14 +103,39 @@ var app = {}, browserSettings = {}, browserStorage = $.localStorage;
     }
   }
 
-  function updateTitle() {
+  //see http://stackoverflow.com/a/9609450
+  var decodeEntities = (function() {
```
Comment (jasoncalabrese) [inline static/js/client.js]: Codacy Issue found: ['decodeEntities' is defined but never used.](https://www.codacy.com/app/Nightscout/cgm-remote-monitor/file/768506908/issues/source?bid=2452379&fileBranchId=2457543#l107)

File: lib/plugins/cannulaage.js [hunk 4]
```diff
@@ -12,40 +12,83 @@ function init() {
...
+  function frac(f) {
+    return f % 1;
+  }
+
+  cage.findLatestTimeChange = function findLatestTimeChange(sbx) {
+  
+  var returnValue = {'message':'', 'found': false, 'age': 0, 'treatmentDate': null, 'checkForAlert': false};
+  
+      _.forEach(sbx.data.treatments, function eachTreatment (treatment) {
+      if (treatment.eventType == 'Site Change') {
```
Comment (jasoncalabrese) [inline lib/plugins/cannulaage.js]: Codacy Issue found: [Expected '===' and instead saw '=='.](https://www.codacy.com/app/Nightscout/cgm-remote-monitor/file/768506884/issues/source?bid=2452379&fileBranchId=2457543#l53)

File: lib/plugins/cannulaage.js [hunk 5]
```diff
@@ -12,40 +12,83 @@ function init() {
...
+          , message: title + ', time to change?'
+          , pushoverSound: 'persistent'
+          , plugin: cage
+          , debug: {
+            cannulaInfo: cannulaInfo
+        }
+      });
+      
+    }
+  }
```
Comment (jasoncalabrese) [inline lib/plugins/cannulaage.js]: Codacy Issue found: [Missing semicolon.](https://www.codacy.com/app/Nightscout/cgm-remote-monitor/file/768506884/issues/source?bid=2452379&fileBranchId=2457543#l42)

File: lib/plugins/cannulaage.js [hunk 6]
```diff
@@ -12,40 +12,83 @@ function init() {
...
-        var hours = Math.round(Math.abs(sbx.time - treatmentDate) / 36e5);
-
-        if (!found) {
-          found = true;
-          age = hours;
+  cage.checkNotifications = function checkNotifications(sbx) {
+  
+    // Just return if no notification times are configured
+  
+  	if (!sbx.extendedSettings.urgent && !sbx.extendedSettings.warn) return;
```
Comment (jasoncalabrese) [inline lib/plugins/cannulaage.js]: Codacy Issue found: [Expected '{' and instead saw 'return'.](https://www.codacy.com/app/Nightscout/cgm-remote-monitor/file/768506884/issues/source?bid=2452379&fileBranchId=2457543#l19)

File: static/js/client.js [hunk 7]
```diff
@@ -124,6 +149,10 @@ var app = {}, browserSettings = {}, browserStorage = $.localStorage;
         bg_title = s(scaleBg(currentMgdl)) + s(deltaDisplay) + s(direction.info(latestSGV).label) + bg_title;
       }
     }
+    } else {
+    	bg_title = message.title + ' ' + message.message;
+    	$('h1.customTitle').text(bg_title);
+    }
 
     $(document).attr('title', bg_title);
```
Comment (jasoncalabrese) [inline static/js/client.js]: Codacy Issue found: ['bg_title' used out of scope.](https://www.codacy.com/app/Nightscout/cgm-remote-monitor/file/768506908/issues/source?bid=2452379&fileBranchId=2457543#l157)

File: static/js/client.js [hunk 8]
```diff
@@ -103,14 +103,39 @@ var app = {}, browserSettings = {}, browserStorage = $.localStorage;
...
+    function s(value, sep) { return value ? value + ' ' : sep || ''; }
 
+	if (!message && !alarmInProgress) {
     var time = latestSGV ? new Date(latestSGV.x).getTime() : (prevSGV ? new Date(prevSGV.x).getTime() : -1)
       , ago = timeAgo(time, browserSettings);
 
     var bg_title = browserSettings.customTitle || '';
 
-    function s(value, sep) { return value ? value + ' ' : sep || ''; }
+    if (browserSettings.customTitle) $('h1.customTitle').text(browserSettings.customTitle);
```
Comment (jasoncalabrese) [inline static/js/client.js]: Codacy Issue found: [Expected '{' and instead saw '$'.](https://www.codacy.com/app/Nightscout/cgm-remote-monitor/file/768506908/issues/source?bid=2452379&fileBranchId=2457543#l138)

File: lib/plugins/cannulaage.js [hunk 9]
```diff
@@ -12,40 +12,83 @@ function init() {
...
+  }
+  
+  cage.updateVisualisation = function updateVisualisation (sbx) {
 
-    var info = [{label: 'Inserted:', value: moment(treatmentDate).format('lll')}];
-    if (message !== '') { info.push({label: 'Notes:', value: message}); }
+    var cannulaInfo = cage.findLatestTimeChange(sbx);
+
+    var info = [{label: 'Inserted:', value: moment(cannulaInfo.treatmentDate).format('lll')}];
+    if (cannulaInfo.message != '') info.push({label: 'Notes:', value: cannulaInfo.message});
```
Comment (jasoncalabrese) [inline lib/plugins/cannulaage.js]: Codacy Issue found: [Expected '!==' and instead saw '!='.](https://www.codacy.com/app/Nightscout/cgm-remote-monitor/file/768506884/issues/source?bid=2452379&fileBranchId=2457543#l82)

**Comment by jasoncalabrese** [comment]
wish codacy didn't comment as me, I need to figure out how to change it

**Comment by Fokko** [comment]
Good point, sometimes it is a bit confusing. :+1:

**Comment by sulkaharo** [comment]
Codacy <3. I'll do these today. :)

**Comment by jasoncalabrese** [comment]
For BWP it might be good to have an info setting too, I don't think I'd want a warning or urgent till it was past time for a site change.  But an info level notification about 4 hours before would be really good.  We should also see if we can find a more distinct sound for this type of notification. Maybe `incoming`?  More at https://pushover.net/api#sounds

File: static/js/client.js (hunk 3 above)
Comment (jasoncalabrese) [inline static/js/client.js]: I was able to remove this when I made direction a plugin, must have been added back with the merge

File: lib/profilefunctions.js (hunk 1 above)
Comment (jasoncalabrese) [inline lib/profilefunctions.js]: Codacy Issue found: [Expected '{' and instead saw 'tz'.](https://www.codacy.com/app/Nightscout/cgm-remote-monitor/file/769371558/issues/source?bid=2452379&fileBranchId=2457543#l41)

File: lib/plugins/cannulaage.js (hunk 2 above)
Comment (jasoncalabrese) [inline lib/plugins/cannulaage.js]: Codacy Issue found: [Missing semicolon.](https://www.codacy.com/app/Nightscout/cgm-remote-monitor/file/769371543/issues/source?bid=2452379&fileBranchId=2457543#l75)

File: static/js/client.js (hunk 3 above)
Comment (jasoncalabrese) [inline static/js/client.js]: Codacy Issue found: ['decodeEntities' is defined but never used.](https://www.codacy.com/app/Nightscout/cgm-remote-monitor/file/769371567/issues/source?bid=2452379&fileBranchId=2457543#l107)

File: lib/plugins/cannulaage.js (hunk 4 above)
Comment (jasoncalabrese) [inline lib/plugins/cannulaage.js]: Codacy Issue found: [Expected '===' and instead saw '=='.](https://www.codacy.com/app/Nightscout/cgm-remote-monitor/file/769371543/issues/source?bid=2452379&fileBranchId=2457543#l53)

File: lib/plugins/cannulaage.js (hunk 5 above)
Comment (jasoncalabrese) [inline lib/plugins/cannulaage.js]: Codacy Issue found: [Missing semicolon.](https://www.codacy.com/app/Nightscout/cgm-remote-monitor/file/769371543/issues/source?bid=2452379&fileBranchId=2457543#l42)

File: lib/plugins/cannulaage.js (hunk 6 above)
Comment (jasoncalabrese) [inline lib/plugins/cannulaage.js]: Codacy Issue found: [Expected '{' and instead saw 'return'.](https://www.codacy.com/app/Nightscout/cgm-remote-monitor/file/769371543/issues/source?bid=2452379&fileBranchId=2457543#l19)

File: static/js/client.js (hunk 7 above)
Comment (jasoncalabrese) [inline static/js/client.js]: Codacy Issue found: ['bg_title' used out of scope.](https://www.codacy.com/app/Nightscout/cgm-remote-monitor/file/769371567/issues/source?bid=2452379&fileBranchId=2457543#l157)

File: static/js/client.js (hunk 8 above)
Comment (jasoncalabrese) [inline static/js/client.js]: Codacy Issue found: [Expected '{' and instead saw '$'.](https://www.codacy.com/app/Nightscout/cgm-remote-monitor/file/769371567/issues/source?bid=2452379&fileBranchId=2457543#l138)

File: lib/plugins/cannulaage.js (hunk 9 above)
Comment (jasoncalabrese) [inline lib/plugins/cannulaage.js]: Codacy Issue found: [Expected '!==' and instead saw '!='.](https://www.codacy.com/app/Nightscout/cgm-remote-monitor/file/769371543/issues/source?bid=2452379&fileBranchId=2457543#l82)

File: static/js/client.js [hunk 10]
```diff
@@ -124,6 +149,10 @@ var app = {}, browserSettings = {}, browserStorage = $.localStorage;
         bg_title = s(scaleBg(currentMgdl)) + s(deltaDisplay) + s(direction.info(latestSGV).label) + bg_title;
       }
     }
+    } else {
+    	bg_title = message.title + ' ' + message.message;
```
Comment (jasoncalabrese) [inline static/js/client.js]: Codacy Issue found: ['bg_title' used out of scope.](https://www.codacy.com/app/Nightscout/cgm-remote-monitor/file/769371567/issues/source?bid=2452379&fileBranchId=2457543#l153)

File: lib/plugins/cannulaage.js [hunk 11]
```diff
@@ -12,40 +12,87 @@ function init() {
...
+  }
+  
+  cage.updateVisualisation = function updateVisualisation (sbx) {
 
-    var info = [{label: 'Inserted:', value: moment(treatmentDate).format('lll')}];
-    if (message !== '') { info.push({label: 'Notes:', value: message}); }
+    var cannulaInfo = cage.findLatestTimeChange(sbx);
+
+    var info = [{label: 'Inserted:', value: moment(cannulaInfo.treatmentDate).format('lll')}];
+    if (cannulaInfo.message !== '') info.push({label: 'Notes:', value: cannulaInfo.message});
```
Comment (jasoncalabrese) [inline lib/plugins/cannulaage.js]: Codacy Issue found: [Expected '{' and instead saw 'info'.](https://www.codacy.com/app/Nightscout/cgm-remote-monitor/file/769393815/issues/source?bid=2452379&fileBranchId=2457543#l86)

File: static/js/client.js [hunk 12]
```diff
@@ -103,26 +103,57 @@ var app = {}, browserSettings = {}, browserStorage = $.localStorage;
     }
   }
 
-  function updateTitle() {
+  //see http://stackoverflow.com/a/9609450
+  var decodeEntities = (function() {
```
Comment (jasoncalabrese) [inline static/js/client.js]: Codacy Issue found: ['decodeEntities' is defined but never used.](https://www.codacy.com/app/Nightscout/cgm-remote-monitor/file/769393839/issues/source?bid=2452379&fileBranchId=2457543#l107)

File: lib/plugins/cannulaage.js [hunk 13]
```diff
@@ -12,40 +12,87 @@ function init() {
...
+          , message: title + ', time to change?'
+          , pushoverSound: 'persistent'
+          , plugin: cage
+          , debug: {
+            cannulaInfo: cannulaInfo
+        }
+      });
+      
+    }
+  }
```
Comment (jasoncalabrese) [inline lib/plugins/cannulaage.js]: Codacy Issue found: [Missing semicolon.](https://www.codacy.com/app/Nightscout/cgm-remote-monitor/file/769393815/issues/source?bid=2452379&fileBranchId=2457543#l46)

File: lib/profilefunctions.js [hunk 14]
```diff
@@ -36,30 +37,43 @@ function init(profileData) {
...
 	};
 	
 	profile.getCurrentProfile = function getCurrentProfile() {
     return profile.hasData() ? profile.data[0] : {};
   };
-	
+
+  profile.getTimezone = function getTimezone() {
+    return profile.getCurrentProfile()['timezone'];
+  }
```
Comment (jasoncalabrese) [inline lib/profilefunctions.js]: Codacy Issue found: [Missing semicolon.](https://www.codacy.com/app/Nightscout/cgm-remote-monitor/file/769393830/issues/source?bid=2452379&fileBranchId=2457543#l75)

File: static/js/client.js [hunk 15]
```diff
@@ -1338,10 +1374,12 @@ var app = {}, browserSettings = {}, browserStorage = $.localStorage;
       currentAlarmType = alarm.type;
       console.info('generating timeAgoAlarm', alarm.type);
       $('#container').addClass('alarming-timeago');
-      if (level === 'warn') {
-        generateAlarm(alarmSound);
+      console.log('ago:', ago);
+      var message = {'title': 'Last data received ', 'message': ago.value + ago.label};
+      if (level == 'warn') {
```
Comment (jasoncalabrese) [inline static/js/client.js]: Codacy Issue found: [Expected '===' and instead saw '=='.](https://www.codacy.com/app/Nightscout/cgm-remote-monitor/file/769393839/issues/source?bid=2452379&fileBranchId=2457543#l1379)

File: static/js/client.js [hunk 16]
```diff
@@ -103,26 +103,57 @@ var app = {}, browserSettings = {}, browserStorage = $.localStorage;
...
+      return str;
+    }
 
-    var time = latestSGV ? new Date(latestSGV.x).getTime() : (prevSGV ? new Date(prevSGV.x).getTime() : -1)
-      , ago = timeAgo(time, browserSettings);
+    return decodeHTMLEntities;
+  })();
 
-    var bg_title = browserSettings.customTitle || '';
+  function updateTitle(message) {
```
Comment (jasoncalabrese) [inline static/js/client.js]: Codacy Issue found: [This function's cyclomatic complexity is too high. (9)](https://www.codacy.com/app/Nightscout/cgm-remote-monitor/file/769393839/issues/source?bid=2452379&fileBranchId=2457543#l127)

File: lib/plugins/cannulaage.js [hunk 17]
```diff
@@ -12,40 +12,87 @@ function init() {
...
-              message = treatment.notes;
+              returnValue.message = treatment.notes;
             }
           }
         }
       }
     });
+    
+    return returnValue;
+  }
```
Comment (jasoncalabrese) [inline lib/plugins/cannulaage.js]: Codacy Issue found: [Missing semicolon.](https://www.codacy.com/app/Nightscout/cgm-remote-monitor/file/769393815/issues/source?bid=2452379&fileBranchId=2457543#l79)

File: lib/plugins/cannulaage.js (hunk 11 above)
Comment (jasoncalabrese) [inline lib/plugins/cannulaage.js]: Codacy Issue found: [Expected '{' and instead saw 'info'.](https://www.codacy.com/app/Nightscout/cgm-remote-monitor/file/769403721/issues/source?bid=2452379&fileBranchId=2457543#l86)

File: static/js/client.js [hunk 18]
```diff
@@ -1338,10 +1360,12 @@ var app = {}, browserSettings = {}, browserStorage = $.localStorage;
       currentAlarmType = alarm.type;
       console.info('generating timeAgoAlarm', alarm.type);
       $('#container').addClass('alarming-timeago');
-      if (level === 'warn') {
-        generateAlarm(alarmSound);
+      console.log('ago:', ago);
+      var message = {'title': 'Last data received ', 'message': ago.value + ago.label};
+      if (level == 'warn') {
```
Comment (jasoncalabrese) [inline static/js/client.js]: Codacy Issue found: [Expected '===' and instead saw '=='.](https://www.codacy.com/app/Nightscout/cgm-remote-monitor/file/769403745/issues/source?bid=2452379&fileBranchId=2457543#l1365)

File: lib/plugins/cannulaage.js (hunk 13 above)
Comment (jasoncalabrese) [inline lib/plugins/cannulaage.js]: Codacy Issue found: [Missing semicolon.](https://www.codacy.com/app/Nightscout/cgm-remote-monitor/file/769403721/issues/source?bid=2452379&fileBranchId=2457543#l46)

File: lib/profilefunctions.js (hunk 14 above)
Comment (jasoncalabrese) [inline lib/profilefunctions.js]: Codacy Issue found: [Missing semicolon.](https://www.codacy.com/app/Nightscout/cgm-remote-monitor/file/769403736/issues/source?bid=2452379&fileBranchId=2457543#l75)

File: lib/plugins/cannulaage.js (hunk 17 above)
Comment (jasoncalabrese) [inline lib/plugins/cannulaage.js]: Codacy Issue found: [Missing semicolon.](https://www.codacy.com/app/Nightscout/cgm-remote-monitor/file/769403721/issues/source?bid=2452379&fileBranchId=2457543#l79)

File: static/js/client.js (hunk 18 above)
Comment (jasoncalabrese) [inline static/js/client.js]: Codacy Issue found: [Expected '===' and instead saw '=='.](https://www.codacy.com/app/Nightscout/cgm-remote-monitor/file/769511252/issues/source?bid=2452379&fileBranchId=2457543#l1365)

File: lib/plugins/cannulaage.js [hunk 19]
```diff
@@ -4,52 +4,96 @@ var _ = require('lodash');
...
+
+  return returnValue;
 }
 
+cage.updateVisualisation = function updateVisualisation (sbx) {
+
+  var cannulaInfo = cage.findLatestTimeChange(sbx);
+
+  var info = [{label: 'Inserted:', value: moment(cannulaInfo.treatmentDate).format('lll')}];
+  if (cannulaInfo.message !== '') info.push({label: 'Notes:', value: cannulaInfo.message});
```
Comment (jasoncalabrese) [inline lib/plugins/cannulaage.js]: Codacy Issue found: [Expected '{' and instead saw 'info'.](https://www.codacy.com/app/Nightscout/cgm-remote-monitor/file/769511228/issues/source?bid=2452379&fileBranchId=2457543#l90)

File: lib/plugins/cannulaage.js [hunk 20]
```diff
@@ -4,52 +4,96 @@ var _ = require('lodash');
...
+          if (treatment.notes) {
+            returnValue.message = treatment.notes;
+          }
+        }
+      }
+    }
+  });
+
+  return returnValue;
 }
```
Comment (jasoncalabrese) [inline lib/plugins/cannulaage.js]: Codacy Issue found: [Missing semicolon.](https://www.codacy.com/app/Nightscout/cgm-remote-monitor/file/769511228/issues/source?bid=2452379&fileBranchId=2457543#l83)

File: lib/plugins/cannulaage.js [hunk 21]
```diff
@@ -4,52 +4,96 @@ var _ = require('lodash');
...
+      , plugin: cage
+      , debug: {
+        cannulaInfo: cannulaInfo
       }
     });
 
-    var info = [{label: 'Inserted:', value: moment(treatmentDate).format('lll')}];
-    if (message !== '') { info.push({label: 'Notes:', value: message}); }
+  }
+}
```
Comment (jasoncalabrese) [inline lib/plugins/cannulaage.js]: Codacy Issue found: [Missing semicolon.](https://www.codacy.com/app/Nightscout/cgm-remote-monitor/file/769511228/issues/source?bid=2452379&fileBranchId=2457543#l50)

File: lib/profilefunctions.js (hunk 14 above)
Comment (jasoncalabrese) [inline lib/profilefunctions.js]: Codacy Issue found: [Missing semicolon.](https://www.codacy.com/app/Nightscout/cgm-remote-monitor/file/769511243/issues/source?bid=2452379&fileBranchId=2457543#l75)

File: static/js/client.js (hunk 18 above)
Comment (jasoncalabrese) [inline static/js/client.js]: Codacy Issue found: [Expected '===' and instead saw '=='.](https://www.codacy.com/app/Nightscout/cgm-remote-monitor/file/769587981/issues/source?bid=2452379&fileBranchId=2457543#l1365)

File: lib/plugins/cannulaage.js [hunk 22]
```diff
@@ -3,53 +3,110 @@
...
         }
       }
-    });
+    }
+  });
 
-    var info = [{label: 'Inserted:', value: moment(treatmentDate).format('lll')}];
-    if (message !== '') { info.push({label: 'Notes:', value: message}); }
+  return returnValue;
+}
```
Comment (jasoncalabrese) [inline lib/plugins/cannulaage.js]: Codacy Issue found: [Missing semicolon.](https://www.codacy.com/app/Nightscout/cgm-remote-monitor/file/769587957/issues/source?bid=2452379&fileBranchId=2457543#l96)

File: lib/plugins/cannulaage.js [hunk 23]
```diff
@@ -3,53 +3,110 @@
...
+      , plugin: cage
+      , debug: {
+        cannulaInfo: cannulaInfo
+      }
+    });
 
-  function cage() {
-    return cage;
   }
+}
```
Comment (jasoncalabrese) [inline lib/plugins/cannulaage.js]: Codacy Issue found: [Missing semicolon.](https://www.codacy.com/app/Nightscout/cgm-remote-monitor/file/769587957/issues/source?bid=2452379&fileBranchId=2457543#l54)

File: lib/profilefunctions.js (hunk 14 above)
Comment (jasoncalabrese) [inline lib/profilefunctions.js]: Codacy Issue found: [Missing semicolon.](https://www.codacy.com/app/Nightscout/cgm-remote-monitor/file/769587972/issues/source?bid=2452379&fileBranchId=2457543#l75)

File: lib/plugins/cannulaage.js [hunk 24]
```diff
@@ -3,53 +3,110 @@
...
 function init() {
+  return cage();
+}
+
+module.exports = init;
+
+cage.label = 'Cannula Age';
+cage.pluginType = 'pill-minor';
+
+cage.checkNotifications = function checkNotifications(sbx) {
```
Comment (jasoncalabrese) [inline lib/plugins/cannulaage.js]: Codacy Issue found: [This function's cyclomatic complexity is too high. (11)](https://www.codacy.com/app/Nightscout/cgm-remote-monitor/file/769587957/issues/source?bid=2452379&fileBranchId=2457543#l19)

File: lib/plugins/cannulaage.js [hunk 25]
```diff
@@ -3,53 +3,110 @@
...
-    });
+cage.updateVisualisation = function updateVisualisation (sbx) {
 
-  };
+  var cannulaInfo = cage.findLatestTimeChange(sbx);
 
-  return cage();
-}
+  var info = [{label: 'Inserted:', value: moment(cannulaInfo.treatmentDate).format('lll')}];
+  if (cannulaInfo.message !== '') info.push({label: 'Notes:', value: cannulaInfo.message});
```
Comment (jasoncalabrese) [inline lib/plugins/cannulaage.js]: Codacy Issue found: [Expected '{' and instead saw 'info'.](https://www.codacy.com/app/Nightscout/cgm-remote-monitor/file/769587957/issues/source?bid=2452379&fileBranchId=2457543#l103)

File: lib/plugins/cannulaage.js [hunk 26]
```diff
@@ -3,53 +3,110 @@
...
+        cannulaInfo: cannulaInfo
+      }
+    });
 
-  function cage() {
-    return cage;
   }
+}
+
+function frac(f) {
```
Comment (jasoncalabrese) [inline lib/plugins/cannulaage.js]: Codacy Issue found: ['frac' is defined but never used.](https://www.codacy.com/app/Nightscout/cgm-remote-monitor/file/769587957/issues/source?bid=2452379&fileBranchId=2457543#l56)

File: static/js/client.js (hunk 18 above)
Comment (jasoncalabrese) [inline static/js/client.js]: Codacy Issue found: [Expected '===' and instead saw '=='.](https://www.codacy.com/app/Nightscout/cgm-remote-monitor/file/769678660/issues/source?bid=2452379&fileBranchId=2457543#l1365)

File: lib/plugins/cannulaage.js [hunk 27]
```diff
@@ -3,53 +3,104 @@
...
 function init() {
+  return cage();
+}
+
+module.exports = init;
+
+cage.label = 'Cannula Age';
+cage.pluginType = 'pill-minor';
+
+cage.checkNotifications = function checkNotifications(sbx) {
```
Comment (jasoncalabrese) [inline lib/plugins/cannulaage.js]: Codacy Issue found: [This function's cyclomatic complexity is too high. (11)](https://www.codacy.com/app/Nightscout/cgm-remote-monitor/file/769678636/issues/source?bid=2452379&fileBranchId=2457543#l19)

Comment (jasoncalabrese) [inline lib/plugins/cannulaage.js]: Codacy Issue found: [This function's cyclomatic complexity is too high. (11)](https://www.codacy.com/app/Nightscout/cgm-remote-monitor/file/769743224/issues/source?bid=2452379&fileBranchId=2457543#l19)

File: lib/plugins/cannulaage.js [hunk 28]
```diff
@@ -3,53 +3,109 @@
...
-  function cage() {
-    return cage;
+cage.getPrefs = function getPrefs(sbx) {
+  // CAGE_INFO = 44 CAGE_WARN=48 CAGE_URGENT=70
+  return {
+    'info' : sbx.extendedSettings.info ? sbx.extendedSettings.info : 44,
+    'warn' : sbx.extendedSettings.warn ? sbx.extendedSettings.warn : 48,
+    'urgent' : sbx.extendedSettings.urgent ? sbx.extendedSettings.urgent : 72,
+    'enableAlerts' : sbx.extendedSettings.enablealerts
   }
```
Comment (jasoncalabrese) [inline lib/plugins/cannulaage.js]: Codacy Issue found: [Missing semicolon.](https://www.codacy.com/app/Nightscout/cgm-remote-monitor/file/769778659/issues/source?bid=2452379&fileBranchId=2457543#l26)

File: lib/plugins/cannulaage.js [hunk 29]
```diff
@@ -3,53 +3,109 @@
...
+}
```
Comment (jasoncalabrese) [inline lib/plugins/cannulaage.js]: Codacy Issue found: [Missing semicolon.](https://www.codacy.com/app/Nightscout/cgm-remote-monitor/file/769778659/issues/source?bid=2452379&fileBranchId=2457543#l27)

**Pull request merged**